internal/core/src/indexbuilder/init_c.cpp

```diff
@@ -23,6 +24,9 @@ char*
 IndexBuilderSetSimdType(const char* value) {
     auto real_type = milvus::config::KnowhereSetSimdType(value);
     char* ret = reinterpret_cast<char*>(malloc(real_type.length() + 1));
+    if (ret == nullptr) {
+        throw std::runtime_error("Error: Memory allocation failed");
```

Comment:
Better to use "AssertInfo" instead.

Reply:
Thanks for the suggestion. Currently working on it.